crates/database/src/events.rs

```diff
@@ -47,6 +47,7 @@ pub async fn last_block(ex: &mut PgConnection) -> Result<i64, sqlx::Error> {
     const QUERY: &str = "\
             SELECT GREATEST( (SELECT COALESCE(MAX(block_number), 0) FROM trades), (SELECT \
                          COALESCE(MAX(block_number), 0) FROM settlements), (SELECT \
+                         COALESCE(MAX(block_number), 0) FROM settlement_observations),(SELECT \
```

Comment:
I added it for completeness, it does not hurt. However, the function itself is not reliable since the tables can be left in an inconsistent state (append and delete functions are not atomic). At least this is my understanding.